TutorialBot/TutorialBotOutput.py

```diff
@@ -0,0 +1,143 @@
+            steer = 0
+
+        if abs(math.degrees(angle_front_to_target)) < POWERSLIDE_ANGLE:
+            powerslide = True
+        else:
+            powerslide = False
+
+    # def check_for_dodge(target_x, target_y):
+    #     if should_dodge and time.time() > next_dodge_time:
+    #         aim(target_x, target_y)
+    #         jump = True
+    #         pitch = -1
+    #
+    #         if on_second_jump:
+    #             on_second_jump = False
+    #             should_dodge = False
+    #         else:
+    #             on_second_jump = True
+    #             next_dodge_time = time.time() + DODGE_TIME
+
+    index = 0
+
+    # Contants
+    DODGE_TIME = 0.2
+    DISTANCE_TO_DODGE = 500
+    DISTANCE_FROM_BALL_TO_BOOST = 1500  # The minimum distance the ball needs to be away from the bot for the bot to boost
+    POWERSLIDE_ANGLE = 170  # The angle (from the front of the bot to the ball) at which the bot should start to powerslide.
+
+    # Controller inputs
+    # throttle = 0 defined in 100
+    steer = 0
+    pitch = 0
+    yaw = 0
+    roll = 0
+    # boost = False
+    jump = False
+    powerslide = False
+
+    # Game values
+    bot_pos = None
+    bot_rot = None
+    ball_pos = None
+    bot_yaw = None
+
+    # Dodging
+    # should_dodge = False
+    # on_second_jump = False
+    # next_dodge_time = 0
+
+    # Update game data variables
+    bot_pos = values.gamecars[index].Location
+    bot_rot = values.gamecars[index].Rotation
+    ball_pos = values.gameball.Location
+
+    # Get car's yaw and convert from Unreal Rotator units to degrees
+    bot_yaw = abs(bot_rot.Yaw) % 65536 / 65536 * 360
+    if bot_rot.Yaw < 0:
+        bot_yaw *= -1
+
+    # Boost when ball is far enough away
+    if distance(bot_pos.X, bot_pos.Y, ball_pos.X, ball_pos.Y) > DISTANCE_FROM_BALL_TO_BOOST:
+        boost = True
+    else:
+        boost = False
+
+    # Blue has their goal at -5000 (Y axis) and orange has their goal at 5000 (Y axis). This means that:
+    # - Blue is behind the ball if the ball's Y axis is greater than blue's Y axis
+    # - Orange is behind the ball if the ball's Y axis is smaller than orange's Y axis
+    throttle = 1
+    if (index == 0 and bot_pos.Y < ball_pos.Y) or (index == 1 and bot_pos.Y > ball_pos.Y):
+        aim(ball_pos.X, ball_pos.Y)
+
+        # if distance(bot_pos.X, bot_pos.Y, ball_pos.X, ball_pos.Y) < DISTANCE_TO_DODGE:
+        #     should_dodge = True
+    else:
+        if index == 0:
+            # Blue team's goal is located at (0, -5000)
+            aim(0, -5000)
+        else:
+            # Orange team's goal is located at (0, 5000)
+            aim(0, 5000)
+
+    # Boost on kickoff
+    if ball_pos.X == 0 and ball_pos.Y == 0:
+        aim(ball_pos.X, ball_pos.Y)
+        boost = True
+        throttle = 1
+
+    # check_for_dodge(ball_pos.X, ball_pos.Y)
+
+    # output = [0] * 8  # Initialised all to neutral
+    output = tf.Variable(0)
```

Comment:
you did this correct but....
This can not be inside this method.
It needs to happen before we start creating the batches and it should happen in this file.